docs: Add the onchain identifier guide (#574)
safe-global/safe-docs: pages/sdk/relay-kit/guides/onchain-identifier.mdx

import { Tabs, Steps, Callout } from 'nextra/components'

# Track ERC-4337 transactions with an onchain identifier

> In this guide, you will learn how to get an identifier and track all your ERC-4337 Safe transactions onchain by attaching it to the user operations `callData`.

Reviewer: ```suggestion
In this guide, you will learn how to get an identifier and track all your ERC-4337 Safe transactions onchain by attaching it to the user operations (`callData`).
```

Reviewer: Should we also link to the User Operations entry in the Glossary?

Reviewer: Re: my suggestion regarding the parenthesis, feel free to ignore if that's not accurate way to represent.

Author: Regarding the parenthesis, this should be the right option:
```
user operation's `callData`.
```

> This guide summarizes all the steps included in the [Safe accounts with the Safe4337Module guide](./4337-safe-sdk.mdx), with an additional step to add the `onchainIdentifier` to your Safe transactions. If you already have the Relay Kit integrated, continue reading to learn how to get your onchain identifier, and feel free to jump straight to [step number 4](#add-your-onchain-identifier-to-the-user-operation-calldata) to learn how to use it afterward.

Reviewer: ```suggestion
This guide summarizes all the steps included in the [Safe accounts with the Safe4337Module guide](./4337-safe-sdk.mdx), with an additional step to add the `onchainIdentifier` to your Safe transactions. If you already have the Relay Kit integrated, continue reading to learn how to get your onchain identifier, and feel free to jump straight to [step number five](#add-the-onchainidentifier-to-the-calldata) to learn how to use it afterward.
```

Author: Good catch!

> Please check the Safe4337Module linked guide for all the details regarding the steps summarized here.

Reviewer: Could potentially remove this statement as it's similar to the first statement of the last paragraph.

  {/* <!-- vale off --> */}

  ```typescript
  const userOperationHash = await safe4337Pack.executeTransaction({
    executable: signedSafeOperation
  })
  ```

  {/* <!-- vale on --> */}

  Once the transaction is executed, you will see the `onchainIdentifier` included at the end of the user operation `callData`. 

  ### Check the transaction status

  To check the transaction status, we can use the `getTransactionReceipt()` method, which returns the transaction receipt after execution.

  {/* <!-- vale off --> */}

  ```typescript
  let userOperationReceipt = null

  while (!userOperationReceipt) {
    // Wait 2 seconds before checking the status again
    await new Promise((resolve) => setTimeout(resolve, 2000))
    userOperationReceipt = await safe4337Pack.getUserOperationReceipt(
      userOperationHash
    )
  }
  ```

  {/* <!-- vale on --> */}

  Now, tracking all your transactions is possible by performing onchain queries looking for your identifier at the end of the `callData` field.

</Steps>

## Recap and further reading

If you are not using the Safe\{Core\} SDK to submit transactions, the following GitHub repository may be helpful:

> - The [safe-onchain-identifiers](https://github.com/5afe/safe-onchain-identifiers) repository showcases where and how to add the identifier at the end of your Safe transactions data without using the Relay Kit and with a low-level detail. Check also the [specific code](https://github.com/5afe/safe-onchain-identifiers/blob/main/test/OnchainIdentifier.ts#L197-L217) where the identifier is concatenated to the `callData`.

Reviewer: ```suggestion
- The [safe-onchain-identifiers](https://github.com/5afe/safe-onchain-identifiers) repository showcases where and how to add the identifier at the end of your Safe transactions data without using the Relay Kit and with a low-level detail. Also check the [code](https://github.com/5afe/safe-onchain-identifiers/blob/main/test/OnchainIdentifier.ts#L197-L217) where the identifier is concatenated to the `callData`.
```
